indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -387,7 +399,7 @@ public LockResult tryLock(final Task task, final LockRequest request)
       if (request instanceof LockRequestForNewSegment) {
         final LockRequestForNewSegment lockRequestForNewSegment = (LockRequestForNewSegment) request;
         if (lockRequestForNewSegment.getGranularity() == LockGranularity.SEGMENT) {
-          newSegmentId = allocateSegmentId(lockRequestForNewSegment, request.getVersion());
+          newSegmentId = allocateSegmentId(lockRequestForNewSegment, request.getVersion(), null);
```

Comment:
why don't we pass in the task group id for segment locks?

Reply:
I suppose we could. My understanding of segment locking is limited which is why I wanted to leave the behaviour intact if possible